plugins/inputs/snmp_trap/snmp_trap_test.go

```diff
@@ -1586,127 +1225,232 @@ func TestReceiveTrapMultipleConfig(t *testing.T) {
+	// the cache and oid lookup is intentionally mocked to fail.
+	trap := gosnmp.SnmpTrap{
+		Variables: []gosnmp.SnmpPDU{
+			{
+				Name:  ".1.3.6.1.2.1.1.3.0",
+				Type:  gosnmp.TimeTicks,
+				Value: now,
+			},
+			{
+				Name:  ".1.3.6.1.6.3.1.1.4.1.0", // SNMPv2-MIB::snmpTrapOID.0
+				Type:  gosnmp.ObjectIdentifier,
+				Value: ".1.3.6.1.6.3.1.1.5.1", // coldStart
+			},
+		},
+	}
 
-			var goSNMP1 gosnmp.GoSNMP
-			var goSNMP2 gosnmp.GoSNMP
-			if tt.version == gosnmp.Version3 {
-				msgFlags := newMsgFlagsV3(tt.secLevel)
-				sp1 := newUsmSecurityParametersForV3(tt.authProto, tt.privProto, tt.secName+"1", tt.privPass+"1", tt.authPass+"1")
-				sp2 := newUsmSecurityParametersForV3(tt.authProto, tt.privProto, tt.secName+"2", tt.privPass+"2", tt.authPass+"2")
-				goSNMP1 = newGoSNMPV3(port1, tt.contextName, tt.engineID, msgFlags, sp1)
-				goSNMP2 = newGoSNMPV3(port2, tt.contextName, tt.engineID, msgFlags, sp2)
-			} else {
-				goSNMP1 = newGoSNMP(tt.version, port1)
-				goSNMP2 = newGoSNMP(tt.version, port2)
-			}
+	// We would prefer to specify port 0 and let the network
+	// stack choose an unused port for us but TrapListener
+	// doesn't have a way to return the autoselected port.
+	// Instead, we'll use an unusual port and hope it's
+	// unused.
+	const port = 12399
 
-			// Send the trap to both receivers
-			sendTrap(t, goSNMP1, tt.trap)
-			sendTrap(t, goSNMP2, tt.trap)
+	// Set up the service input plugin
+	logger := &testutil.CaptureLogger{}
+	plugin := &SnmpTrap{
+		ServiceAddress: "udp://:" + strconv.Itoa(port),
+		Version:        "2c",
+		Translator:     "netsnmp",
+		Log:            logger,
+		timeFunc:       time.Now,
+	}
+	require.NoError(t, plugin.Init())
 
-			// Wait for trap to be received on receiver1
-			select {
-			case <-received1:
-			case <-time.After(2 * time.Second):
-				t.Fatal("timed out waiting for trap to be received")
-			}
-			// Wait for trap to be received on receiver2
-			select {
-			case <-received2:
-			case <-time.After(2 * time.Second):
-				t.Fatal("timed out waiting for trap to be received")
-			}
+	// inject test translator
+	plugin.transl = &testTranslator{}
 
-			// Verify plugin output
-			testutil.RequireMetricsEqual(t,
-				tt.metrics, acc.GetTelegrafMetrics(),
-				testutil.SortMetrics())
-		})
+	var acc testutil.Accumulator
+	require.NoError(t, plugin.Start(&acc))
+	defer plugin.Stop()
+
+	// Create a v1 client and send the trap
+	client := &gosnmp.GoSNMP{
+		Port:      port,
+		Version:   gosnmp.Version2c,
+		Timeout:   2 * time.Second,
+		Retries:   1,
+		MaxOids:   gosnmp.MaxOids,
+		Target:    "127.0.0.1",
+		Community: "public",
+	}
+	require.NoError(t, client.Connect(), "connecting failed")
+	defer client.Conn.Close()
+	_, err := client.SendTrap(trap)
+	require.NoError(t, err, "sending failed")
+	require.NoError(t, client.Conn.Close(), "closing failed")
+
+	// Wait for trap to be received
+	time.Sleep(500 * time.Millisecond)
```

Comment:
use `require.Eventually` instead of `time.sleep`

Reply:
What should you check for if you _don't_ want to receive anything? ;-)

Reply:
Pushed my attempt. Please have a look!

Reply:
Nice use of `Select` statement here, looks perfect!